src/azure-cli/azure/cli/command_modules/appconfig/_diff_utils.py

```diff
@@ -52,7 +52,7 @@ def compare(self, dest_kvs, strict=False):
                 kv_diff[JsonDiff.ADD].append(kv)
 
             else:
-                if not self._kv_equals(kv, dest_kv_lookup[lookup_key], self._compare_fields):
+                if not (ignore_matching_kvs and self._kv_equals(kv, dest_kv_lookup[lookup_key], self._compare_fields)):
```

Comment:
if import mode is all, we dont need to check for updates?

Reply:
Since we will be overwriting the existing kvs anyways, if a kv already exists, my thought was to consider that an update even if it is unchanged